src/verifier/tdx/mod.rs

```diff
@@ -0,0 +1,170 @@
+            hex::encode(&hash_of_nonce_pubkey)
+        );
+
+        verify_evidence(hash_of_nonce_pubkey, &tdx_evidence)
+            .await
+            .map_err(|e| anyhow!("TDX Verifier: {:?}", e))
+    }
+}
+
+#[allow(unused_assignments)]
+async fn verify_evidence(
+    hash_of_nonce_pubkey: Vec<u8>,
+    evidence: &TdxEvidence,
+) -> Result<TeeEvidenceParsedClaim> {
+    // Verify TD quote ECDSA signature.
+    let quote_bin = base64::decode(evidence.quote.clone())?;
+    ecdsa_quote_verification(quote_bin.as_slice()).await?;
+
+    // Parse quote and Compare report data
+    let quote = parse_tdx_quote(&quote_bin)?;
+
+    log::info!("{}\n", &quote);
+
+    if hash_of_nonce_pubkey != quote.report_body.report_data.to_vec() {
+        return Err(anyhow!(
+            "HASH(nonce||pubkey) is different from that in TDX Quote"
+        ));
+    }
+
+    // Verify Integrity of CC Eventlog
+    let mut ccel_data = Vec::default();
+    let mut ccel_option = Option::default();
+    match &evidence.cc_eventlog {
+        Some(el) => {
+            ccel_data = base64::decode(el)?;
+            let reader = CcEventLogReader::new(ccel_data.as_slice())
+                .ok_or_else(|| anyhow!("Parse CC Eventlog failed"))?;
+            let ccel = CcEventLog {
+                cc_events: reader.cc_events,
+            };
+            ccel_option = Some(ccel.clone());
+
+            log::debug!("Get CC Eventlog. \n{}\n", &ccel);
+
+            let rtmr_from_quote = Rtmr {
+                rtmr0: quote.report_body.rtmr_0,
+                rtmr1: quote.report_body.rtmr_1,
+                rtmr2: quote.report_body.rtmr_2,
+                rtmr3: quote.report_body.rtmr_3,
+            };
+
+            ccel.integrity_check(rtmr_from_quote)?;
+        }
+        None => {
+            warn!("There is no CC EventLog in Evidence!!!");
+        }
+    }
+
+    // Return Evidence parsed claim
+    generate_parsed_claim(quote, ccel_option)
+}
+
+fn generate_parsed_claim(
+    quote: Quote,
+    cc_eventlog: Option<CcEventLog>,
+) -> Result<TeeEvidenceParsedClaim> {
+    // Current Parsed Claim just a example
+    // TODO: Claim key name shall be consistent with RVPS
+
+    let mut claim_map = Map::new();
+
+    // Claims from TD Quote.
+    claim_map.insert(
+        "tdx-tcb-svn".to_string(),
+        serde_json::Value::String(hex::encode(quote.report_body.tcb_svn)),
+    );
+    claim_map.insert(
+        "tdx-mrseam".to_string(),
+        serde_json::Value::String(hex::encode(quote.report_body.mr_seam)),
+    );
+    claim_map.insert(
+        "tdx-mrtd".to_string(),
+        serde_json::Value::String(hex::encode(quote.report_body.mr_td)),
+    );
+    claim_map.insert(
+        "tdx-mrconfigid".to_string(),
+        serde_json::Value::String(hex::encode(quote.report_body.mr_config_id)),
+    );
+
+    // Claims from CC EventLog.
+    match cc_eventlog {
+        Some(ccel) => {
+            match ccel.query_digest(MeasuredEntity::TdPayload) {
```

Comment:
Some question from my point of view. Let's take kernel for example. Here the query is based on a principle that the desired CcEvent of the entity will have `11 | "td_payload\0"` as the first bytes of the `data` of the `CcEvent`.

However, when the length of the hashed content of the kernel (s.t. `payload` in td-shim) varies, the expected measurement should also vary. Thus I'd suggest that we add more information about the measured entity, s.t. we replace the key here
```
"tdx-kernel": "<sha384-of-the-evidence>"
```
with 
```
"tdx-kernel": 
    {
    "value": "<sha384-of-the-evidence>",
    "size": "<KERNEL_SIZE-from-CcEvent-data>"
    }
```
or 
```
"tdx-kernel-<KERNEL_SIZE-from-CcEvent-data>": "<sha384-of-the-evidence>"
```

to help get reference data from rvps when the size is different.